src/libraries/Common/src/Interop/Unix/System.Native/Interop.ReadLink.cs

```diff
@@ -66,5 +75,35 @@ internal static partial class Sys
                 outputBufferSize = buffer.Length * 2;
             }
         }
+
+        private static unsafe string? InternalReadLink(ref byte path, ref byte buffer, int bufferSize, ref bool bufferToResize)
+        {
+            int resultLength = ReadLink(
+                        ref path,
+                        ref buffer,
+                        bufferSize);
+
+            if (resultLength < 0)
+            {
+                Interop.Error error = Interop.Sys.GetLastError();
```

Comment:
@pedrobsaila the full error is:

```
/__w/1/s/src/libraries/Common/src/Interop/Unix/System.Native/Interop.ReadLink.cs(94,31): error CS0117: 'Interop' does not contain a definition for 'GetExceptionForIoErrno' [/__w/1/s/src/libraries/System.Net.Ping/src/System.Net.Ping.csproj]
```

You'll have to add the cs file that defines `GetLastError()`, `GetExceptionForIoErrno()` and `Error` to the `System.Net.Ping.csproj` project.